File: GoodFood.Recipes.Api/GoodFood.Recipes.Domain/Model/Recipe.cs
using System;
using System.Collections.Generic;

namespace GoodFood.Recipes.Domain.Model
{
    public class Recipe
    {
        public Guid Id { get; set; }

        public string Description { get; set; }

        public string Slug { get; set; }

        public string Title { get; set; }

        public IEnumerable<RecipeIngredient> RecipeIngredients { get; set; }

        public Guid RecipeCategoryId { get; set; }

        public RecipeCategory RecipeCategory { get; set; }

        public string AppUserId { get; set; }
    }
}

File: GoodFood.Recipes.Client/client-app/src/app/models/recipe.ts
import { IRecipeIngredient } from './recipeIngredient';
import { IRecipeCategory } from './recipeCategory';

export interface IRecipe {
  id: string;
  title: string;
  slug: string;
  description: string;
  recipeCategoryId: string;
  recipeCategory: IRecipeCategory;
  recipeIngredients: IRecipeIngredient[];
}